app/src/main/java/com/keylesspalace/tusky/components/viewthread/ViewThreadViewModel.kt

```diff
@@ -95,36 +98,49 @@ class ViewThreadViewModel @Inject constructor(
     }
 
     fun loadThread(id: String) {
+        _uiState.value = ThreadUiState.Loading
+
         viewModelScope.launch {
+            Log.d(TAG, "Finding status with: $id")
             val contextCall = async { api.statusContext(id) }
-            val statusCall = async { api.statusAsync(id) }
-
-            val contextResult = contextCall.await()
-            val statusResult = statusCall.await()
+            val timelineStatus = db.timelineDao().getStatus(id)
 
-            val status = statusResult.getOrElse { exception ->
-                _uiState.value = ThreadUiState.Error(exception)
-                return@launch
+            val detailedStatus = if (timelineStatus != null) {
+                Log.d(TAG, "Loaded status from local timeline")
+                val viewDatum = timelineStatus.toViewData(gson) as StatusViewData.Concrete
+                viewDatum.copy(isDetailed = true)
+            } else {
+                Log.d(TAG, "Loaded status from network")
+                val statusCall = async { api.statusAsync(id) }
+                val statusResult = statusCall.await()
+                val result = statusResult.getOrElse { exception ->
+                    _uiState.value = ThreadUiState.Error(exception)
+                    return@launch
+                }
+                result.toViewData(true)
             }
```

Comment:
Status from database should only be used to have something to display faster. Even if it is in the database we need to load it from the network to have up to date state (fav/boost count, edited, was it deleted maybe?). After loading it from network it should be saved to the database again so it has the newest state when opening it again afterwards.

Reply:
Are you sure?

By analogy, suppose I've refreshed the home timeline and am scrolling to read the newly loaded statuses.

Your suggestion is like saying that when a new status is scrolled in to view there should be a background refresh of its content just in case it's been edited or deleted in the time that elapsed between the initial refresh and the user scrolling to it.

If the user wants a full refresh of a thread they can swipe-to-refresh (I'll add a menu option for that too as part of https://github.com/tuskyapp/Tusky/pull/3121).

So wouldn't it be better to make the choice to fetch from the network even if it's in the database if `_uiState.value == ThreadUiState.Refreshing`? That's true if the user is refreshing, false otherwise.

Reply:
Yes I think it should refresh when opening the details.
Mostly because I'm used to seeing the most recent information when opening the details. I'm sure other long term Tusky users are as well. Having to manually refresh when it used to be automatic feels weird, especially if replies are always refreshed, why wouldn't the main content.

> Your suggestion is like saying that when a new status is scrolled in to view there should be a background refresh 

No, scolling past a already loaded status and actively clicking on it are two different things.